### PR TITLE
[INTLY-2345] Add openshift templates to walkthrough

### PR DESCRIPTION
**Pull request by dimitraz**
### Description 
Add openshift templates for provisioning spring boot and node.js applications for walkthrough 1A to the `walkthrough.json`

Jira: https://issues.jboss.org/browse/INTLY-2345
Webapp pr: https://github.com/integr8ly/tutorial-web-app/pull/496

### CODE REVIEW

**Comment by pwright** [comment]
@aidenkeating  should these changes live on an OS4 branch until we migrate to OS4?

**Comment by aidenkeating** [comment]
@finp There are changes being made to the webapp to support this. It shouldn't have any impact on how the walkthroughs themselves work as the ServiceInstance version effectively just deployed the template anyway. So I don't think it needs to be kept in a feature branch, unless I'm missing something?

**Comment by pwright** [comment]
just need to fix the test then?
image

File: walkthroughs/1A-integrate-event-and-api-driven-apps/walkthrough.json [hunk 1]
```diff
@@ -9,30 +9,652 @@
...
+                "app": "nodejs-messaging-work-queue-frontend"
+              },
+              "name": "frontend"
+            },
+            "spec": {
+              "port": {
+                "targetPort": 8080
+              },
+              "to": {
+                "kind": "Service"
```
Comment (matskiv) [inline walkthroughs/1A-integrate-event-and-api-driven-apps/walkthrough.json]: This route is missing service name (`nodejs-messaging-work-queue-frontend`), so it doesn't get created.

File: walkthroughs/1A-integrate-event-and-api-driven-apps/walkthrough.json [hunk 2]
```diff
@@ -9,30 +9,652 @@
...
-        "spec": {
-          "clusterServiceClassExternalName": "spring-boot-rest-http-crud",
-          "clusterServicePlanExternalName": "default"
-        }
+        "parameters": [
+          {
+            "name": "RUNTIME_VERSION",
+            "displayName": "OpenJDK 8 image version to use",
+            "description": "Specifies which version of the OpenShift OpenJDK 8 image to use",
+            "value": "1.3-8",
```
Comment (matskiv) [inline walkthroughs/1A-integrate-event-and-api-driven-apps/walkthrough.json:33]: This image tag is more than a year old, CVE grade: F, perhaps we should upgrade? (just an idea, upgrade would be out of scope of this PR)

**Review by matskiv — CHANGES_REQUESTED** [review]
Missing route needs to be fixed. Otherwise looks good.

**Comment by pwright** [comment]
@dimitraz @aidenkeating  am I correct in saying that this PR is now necessary to use web-app master?

**Comment by aidenkeating** [comment]
@finp It shouldn't be required. There have been no changes merged to master to even support this option yet.

When it is merged, it should be backwards compatible.

File: walkthroughs/1A-integrate-event-and-api-driven-apps/walkthrough.json (hunk 1 above)
Comment (dimitraz) [reply walkthroughs/1A-integrate-event-and-api-driven-apps/walkthrough.json]: Good catch thanks @matskiv

**Review by pwright — APPROVED** [review]
Given that the old syntax can still be used, this requires no update to existing documentation at https://access.redhat.com/documentation/en-us/red_hat_managed_integration/1/html-single/getting_started/index#adding_custom_services_to_a_red_hat_managed_integration_walkthrough

File: walkthroughs/1A-integrate-event-and-api-driven-apps/walkthrough.json [hunk 3]
```diff
@@ -9,30 +9,652 @@
...
+                "targetPort": 8080
+              },
+              "to": {
+                "kind": "Service"
+              }
+            }
           }
-        }
+        ],
+        "name": "nodejs-messaging-work-queue-frontend"
```
Comment (matskiv) [inline walkthroughs/1A-integrate-event-and-api-driven-apps/walkthrough.json]: This was added in the wrong place. It needs to be in `spec.to` of the route object. Example:
```
spec:
  host: frontend-admin-tutorial-f6e1.apps.omatskiv-5d7d.openshiftworkshop.com
  wildcardPolicy: None
  port:
    targetPort: 8080
  to:
    kind: Service
    name: nodejs-messaging-work-queue-frontend
```

**Review by matskiv — CHANGES_REQUESTED** [review]

File: walkthroughs/1A-integrate-event-and-api-driven-apps/walkthrough.json (hunk 3 above)
Comment (dimitraz) [reply walkthroughs/1A-integrate-event-and-api-driven-apps/walkthrough.json]: Thanks @matskiv busy looking into this now! Looks like there was an issue when I converted the yaml to json

**Review by matskiv — APPROVED** [review]
I verified with a fix, works well 👍

**Pull request merged**